include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -285,17 +289,20 @@ namespace realization {
              * 
              * @return std::string of the output root directory
              */
-            std::string get_output_root() const noexcept {
+            std::string get_output_root() const {
                 const auto output_root = this->tree.get_optional<std::string>("output_root");
-                if (output_root != boost::none && *output_root != "") {
-                    // Check if the path ends with a trailing slash,
-                    // otherwise add it.
-                    return output_root->back() == '/'
-                           ? *output_root
-                           : *output_root + "/";
-                }
+                const std::string str = *output_root;
```

Comment:
This is potentially UB I believe, since `output_root` may be uninitialized. Need to keep the check on `boost::none` at least, or replace `this->tree.get_optional<std::string>` with a different `get*` call.

Reply:
I'd probably just replace line 293 with this:

```cpp
std::string output_root = this->tree.get<std::string>("output_root");
```

Since this will throw if `output_root` doesn't exist, and removes the need for a separate check.

Phil's suggestion ensures the path references a directory, but it won't handle the trailing slash case.

Reply:
I have my proposed version ready adn tested. But will try this one.

Reply:
This approach doesn't work when `output_rout` doesn't have a "/" at the end. So I kept the check.

Reply:
The problem is when you do have a directory. Instead of writing to the spefied directory, it write to the project directory with the concatenation you pointed out.